### PR TITLE
N-first commit

### PR DESCRIPTION
**Pull request by lNamelessl**
@Grey-A go easy :)

### CODE REVIEW

File: alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+revision: str = "67e690c96c2a"
+down_revision: Union[str, None] = None
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.create_table(
+        "users",
+        sa.Column("id", sa.Integer(), primary_key=True, autoincrement=True),
```
Comment (Grey-A) [inline alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py]: For Aesthetic purposes can you remove the () 👀

File: alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        sa.Column(
+            "profile_picture_url",
+            sa.String(),
+            server_default="/default_profile.jpg",
+            nullable=False,
+        ),
+        sa.Column("full_name", sa.String(50), nullable=False),
+        sa.Column("email", sa.String(), unique=True, index=True, nullable=False),
```
Comment (Grey-A) [inline alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py]: Unique fields are automatically indexed

File: alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    op.create_table(
+        "users",
+        sa.Column("id", sa.Integer(), primary_key=True, autoincrement=True),
+        sa.Column(
+            "profile_picture_url",
+            sa.String(),
+            server_default="/default_profile.jpg",
+            nullable=False,
+        ),
+        sa.Column("full_name", sa.String(50), nullable=False),
```
Comment (Grey-A) [inline alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py]: Be more lenient and use 255

File: alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        sa.Column("email", sa.String(), unique=True, index=True, nullable=False),
+        sa.Column("exception_alert_email", sa.String(), unique=True, nullable=True),
```
Comment (Grey-A) [inline alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py]: Exception alert emails don't need to be unique

File: alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py
Comment (Grey-A) [inline alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py:1]: Can you not put the () on column type declarations when they are empty

File: app/user/apis.py [hunk 5]
```diff
@@ -1,3 +1,95 @@
...
+
+    if credential_in.remember_me:
+        expire_in = settings.REFRESH_TOKEN_EXPIRE_HOURS_LONG
+    else:
+        expire_in = settings.REFRESH_TOKEN_EXPIRE_HOURS
+
+    refresh_token = security.generate_user_token(
+        token_type="refresh",
+        sub=f"USER-{user.id}",
+        expire_in=settings.ACCESS_TOKEN_EXPIRE_MINUTES,
```
Comment (Grey-A) [inline app/user/apis.py]: Use expire_in here

File: app/user/apis.py [hunk 6]
```diff
@@ -1,3 +1,95 @@
...
+    )
+    access_token = security.generate_user_token(
+        token_type="access", sub=f"USER-{user.id}", expire_in=expire_in
```
Comment (Grey-A) [inline app/user/apis.py]: Use access token expire here

File: app/user/schemas/create_schemas.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
+from pydantic import BaseModel, EmailStr, Field
+
+
+class UserCreate(BaseModel):
+    """The User creation model"""
+
+    profile_picture_url: str = Field(
```
Comment (Grey-A) [inline app/user/schemas/create_schemas.py]: No need to add this whole creating, if you check the design when creating user there isn't a field to upload image

File: app/user/schemas/create_schemas.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        description="The url of the image",
+        min_length=1,
+        examples=["https://example.com/image.jpg"],
+    )
+    full_name: str = Field(
+        description="The user's first name",
```
Comment (Grey-A) [inline app/user/schemas/create_schemas.py]: The user's full name

File: app/user/schemas/create_schemas.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        min_length=1,
+        max_length=50,
```
Comment (Grey-A) [inline app/user/schemas/create_schemas.py]: 255 👀

File: app/user/schemas/edit_schemas.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+    email: EmailStr = Field(
+        description="The user's new email",
+        min_length=1,
+        examples=["user@shipnlogic.com"],
+    )
+    edit_token: str = Field(description="The user's edit token", min_length=1)
+
+
+class UserExceptionAlertEmailChange(BaseModel):
```
Comment (Grey-A) [inline app/user/schemas/edit_schemas.py]: No need for a check when editing user exception email

**Review by Grey-A — CHANGES_REQUESTED** [review]
Great job comrade, stay sharp

File: alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py (hunk 2 above)
Comment (lNamelessl) [reply alembic/versions/67e690c96c2a_2024_05_30-created_users_table.py]: OH

File: app/user/schemas/create_schemas.py (hunk 7 above)
Comment (lNamelessl) [reply app/user/schemas/create_schemas.py]: There isn't during sign up but it's in the app

Comment (Grey-A) [reply app/user/schemas/create_schemas.py]: UserCreate is used for signup na?

File: app/user/apis.py (hunk 5 above)
Comment (lNamelessl) [reply app/user/apis.py]: What?

**Pull request merged**